napari/layers/utils/stack_utils.py

```diff
@@ -0,0 +1,188 @@
+    iterable_kwargs = {'scale', 'translate', 'contrast_limits', 'metadata'}
+
+    # turn the kwargs dict into a mapping of {key: iterator}
+    # so that we can use {k: next(v) for k, v in kwargs.items()} below
+    for key, val in kwargs.items():
+        if key == 'colormap' and val is None:
+            if n_channels == 1:
+                kwargs[key] = iter(['gray'])
+            elif n_channels == 2:
+                kwargs[key] = iter(colormaps.MAGENTA_GREEN)
+            else:
+                kwargs[key] = itertools.cycle(colormaps.CYMRGB)
+
+        # make sure that iterable_kwargs are a *sequence* of iterables
+        # for the multichannel case.  For example: if scale == (1, 2) &
+        # n_channels = 3, then scale should == [(1, 2), (1, 2), (1, 2)]
+        elif key in iterable_kwargs:
+            kwargs[key] = iter(ensure_sequence_of_iterables(val, n_channels))
+        else:
+            kwargs[key] = iter(ensure_iterable(val))
+
+    layerdata_list = list()
+    for i in range(n_channels):
+        if multiscale:
+            image = [
+                np.take(data[j], i, axis=channel_axis)
+                for j in range(len(data))
+            ]
+        else:
+            image = np.take(data, i, axis=channel_axis)
+        i_kwargs = {}
+        for key, val in kwargs.items():
+            try:
+                i_kwargs[key] = next(val)
+            except StopIteration:
+                raise IndexError(
+                    "Error adding multichannel image with data shape "
+                    f"{data.shape!r}.\nRequested channel_axis "
+                    f"({channel_axis}) had length {n_channels}, but "
+                    f"the '{key}' argument only provided {i} values. "
+                )
+
+        layerdata = (image, i_kwargs, 'image')
+        layerdata_list.append(layerdata)
+
+    return layerdata_list
+
+
+def stack_to_images(stack: Image, axis: int, **kwargs: Dict,) -> List[Image]:
+    """Function to split the active layer into separate layers along an axis
+
+    Parameters
+    ----------
+    stack : napari.layers.Image
+        The image stack to be split into a list of image layers
+    axis : int
+        The axis to split along.
+
+    Returns
+    -------
+    imagelist: list
+        List of Image objects
+    """
+
+    data, meta, _ = stack.as_layer_data_tuple()
+
+    for key in ("contrast_limits", "colormap", "blending"):
+        meta[key] = None
+
+    name = stack.name
+    num_dim = len(data.shape)
+    # n_channels = data.shape[axis]
+
+    if num_dim < 3:
+        warnings.warn(
+            "The image needs more than 2 dimensions for splitting",
+            UserWarning,
+        )
+        return None
+
+    if axis >= num_dim:
+        warnings.warn(
+            "Can't split along axis {}. The image has {} dimensions".format(
+                axis, num_dim
+            ),
+            UserWarning,
+        )
+        return None
```

Comment:
I think these should raise exceptions rather than warnings.  Otherwise the return type is incorrect and should be changed.  Alternatively, remove this altogether and let the `split_channels_function` below raise the exception (which it will if they are out of bounds)